DNN Platform/Website/DesktopModules/Admin/Authentication/Login.ascx.cs

```diff
@@ -796,10 +798,21 @@ private static string ReplaceLanguage(string Url, string originalLanguage, strin
             return returnValue;
         }
 
+        private string GetLoginPath()
+        {
+            if (this.PortalSettings.LoginTabId == Null.NullInteger)
+            {
+                return LOGIN_PATH;
+            }
+
+            var tab = TabController.Instance.GetTab(this.PortalSettings.LoginTabId, this.PortalId);
+            return tab != null ? new Uri(this._navigationManager.NavigateURL(tab.TabName)).LocalPath : LOGIN_PATH;

```

Comment:
[`NavigateURL(string)`](https://dnndocs.com/api/DotNetNuke.Common.Globals.html#DotNetNuke_Common_Globals_NavigateURL_System_String_) takes a control key, not a tab name, which is why the generated URL is not correct.
```suggestion
            return tab != null ? new Uri(this._navigationManager.NavigateURL(this.PortalSettings.LoginTabId), UriKind.RelativeOrAbsolute).LocalPath : LOGIN_PATH;
```

Reply:
Applied suggestion and reverted comparison